### PR TITLE
`./cg/*.py` – format, lint, & remove docstring spaces

### PR DESCRIPTION
**Pull request by jGaboardi**
xref #589 

This PR formats, lints, & [removes docstring spaces](https://github.com/pysal/libpysal/pull/607#discussion_r1371335093) from `./cg/*.py`. Even though I am trying to do this in coherent sections (left out all `cg` subdirs) some of the PRs are going to be pains to review. I can split into file level PRs if desired.

### CODE REVIEW

File: libpysal/cg/alpha_shapes.py [hunk 1]
```diff
@@ -464,19 +422,17 @@ def alpha_shape(xys, alpha):
...
     ----------
-
     Edelsbrunner, H., Kirkpatrick, D., & Seidel, R. (1983). On the shape of
         a set of points in the plane. IEEE Transactions on information theory,
         29(4), 551-559.
-
     """
     if not HAS_JIT:
-        warn(NUMBA_WARN)
+        warn(NUMBA_WARN)  # noqa B028
```
Comment (martinfleis) [inline libpysal/cg/alpha_shapes.py:432]: Why don't we set `stacklevel` here?

File: libpysal/cg/alpha_shapes.py [hunk 2]
```diff
@@ -571,14 +517,13 @@ def alpha_shape_auto(
...
     ----------
-
     Edelsbrunner, H., Kirkpatrick, D., & Seidel, R. (1983). On the shape of
         a set of points in the plane. IEEE Transactions on information theory,
         29(4), 551-559.
 
     """
     if not HAS_JIT:
-        warn(NUMBA_WARN)
+        warn(NUMBA_WARN)  # noqa B028
```
Comment (martinfleis) [inline libpysal/cg/alpha_shapes.py:526]: As above

File: libpysal/cg/alpha_shapes.py [hunk 3]
```diff
@@ -497,32 +453,24 @@ def _valid_hull(geoms, points):
...
     flag : bool
         Valid hull for alpha shape [True] or not [False]
-
     """
-    flag = True
     # if there is not exactly one polygon
     if geoms.shape[0] != 1:
         return False
     # if any (xys) points do not intersect the polygon
     if HAS_SHAPELY:
```
Comment (martinfleis) [inline libpysal/cg/alpha_shapes.py:470]: Not in this PR but now that we depend on shapely directly in graph, all these if/else paths are obsolete.

File: libpysal/cg/rtree.py [hunk 4]
```diff
@@ -1223,7 +1179,7 @@ def k_means_cluster(root, k, nodes):
     # Initialize: take n random nodes.
     # random.shuffle(ns)
 
-    cluster_starts = ns[:k]
+    ns[:k]
```
Comment (martinfleis) [inline libpysal/cg/rtree.py:1182]: What is this?

File: libpysal/cg/segmentLocator.py [hunk 5]
```diff
@@ -378,17 +379,17 @@ def grid_check(bins, segments, qpoints, visualize=False):
...
     print("Created KDTree in %0.4f seconds" % (t2 - t1))
     if visualize:
-        i = pylab.matshow(
+        pylab.matshow(
             G.grid.mask, origin="lower", extent=G.grid.x_range + G.grid.y_range
         )
 
     t2 = time.time()
-    q = list(map(G.nearest, qpoints))
+    list(map(G.nearest, qpoints))
```
Comment (martinfleis) [inline libpysal/cg/segmentLocator.py:392]: This does not do anything or no?

**Review by martinfleis — COMMENTED** [review]
Yeah, maybe shorter diffs would be better for reviews.

**Comment by jGaboardi** [comment]
> Yeah, maybe shorter diffs would be better for reviews.

How about smaller diffs for future PRs and keep this as is? Is that a fair comprise?

**Comment by martinfleis** [comment]
Yeah for sure, I already went through all of this :D

File: libpysal/cg/segmentLocator.py (hunk 5 above)
Comment (jGaboardi) [reply libpysal/cg/segmentLocator.py:392]: No idea, that must have gotten "fixed" in an autofix...

File: libpysal/cg/rtree.py (hunk 4 above)
Comment (jGaboardi) [reply libpysal/cg/rtree.py:1182]: I have no idea what's going here.

**Comment by jGaboardi** [comment]
@martinfleis I think we should close this PR until I figure out what's going on with `ruff`. There are some very strange changes in here now that I see it.

**Comment by jGaboardi** [comment]
> Yeah for sure, I already went through all of this :D

Oh, you are already done? Curious about those strange changes though.

**Comment by martinfleis** [comment]
> Curious about those strange changes though.

somehow the test are fine with that :D. Can you revert and check which errors it shows?

File: libpysal/cg/alpha_shapes.py (hunk 1 above)
Comment (jGaboardi) [reply libpysal/cg/alpha_shapes.py:432]: Last night it was very close to bedtime when I got this in. I tend to get grumpy when trying to figure out appropriate `stacklevel`. I'll take another look later.

File: libpysal/cg/alpha_shapes.py (hunk 3 above)
Comment (jGaboardi) [reply libpysal/cg/alpha_shapes.py:470]: open an issue for that no – xref #609

File: libpysal/cg/segmentLocator.py (hunk 5 above)
Comment (jGaboardi) [reply libpysal/cg/segmentLocator.py:392]: Ok, so this one is an follows (when also removing the `#noqa B018` from line 382):

```
libpysal/cg/segmentLocator.py:382:5: B018 Found useless expression. Either assign it to a variable or remove it.
libpysal/cg/segmentLocator.py:392:5: F841 [*] Local variable `q` is assigned to but never used
```

Here `ruff`'s autofix altered line 392, but not line 382. So I added `B018` to 382. This is within `grid_check()`, a diagnostic timing function where these variables don't actually need to be stored.

Comment (jGaboardi) [reply libpysal/cg/segmentLocator.py:392]: Let's keep as is in this PR.

File: libpysal/cg/rtree.py (hunk 4 above)
Comment (jGaboardi) [reply libpysal/cg/rtree.py:1182]: OK, here `cluster_starts` is not used anywhere... 

```
libpysal/cg/rtree.py:1182:5: F841 [*] Local variable `cluster_starts` is assigned to but never used
```

However, the `ns[:k]` value *is* being used in the next line. Going to fix and push back up.

**Review by martinfleis — APPROVED** [review]
Alright. Seeing all this, we should probably try to move ahead with actual deprecation of most of the `cg` sooner than later.

**Comment by jGaboardi** [comment]
> Alright. Seeing all this, we should probably try to move ahead with actual deprecation of most of the `cg` sooner than later.

xref #https://github.com/pysal/libpysal/issues/484

**Pull request merged**